cfgov/wellbeing/views.py

```diff
@@ -1,19 +1,20 @@
 from collections import OrderedDict
 
 from django.shortcuts import render
-from django.views.generic import TemplateView
+from django.utils.translation import gettext as _
 
+from core.views import TranslatedTemplateView
 from wellbeing.forms import FWBScore, ResultsForm
 
 
-class ResultsView(TemplateView):
+class ResultsView(TranslatedTemplateView):
     template_name = 'wellbeing/results.html'
 
     # for simplicity, answers below are reordered in order of worst to best
     QUESTIONS = [
         {
             # Question 1
-            'question': 'I could handle a major unexpected expense',
+            'question': _('I could handle a major unexpected expense'),
```

Comment:
Do the rest of the questions (and the answers) in this data structure need to be translated?

Does this one even need to be? Looks like the values in this structure get translated when being output on the results template: https://github.com/cfpb/cfgov-refresh/pull/5668/files#diff-13377cbaabf59c01cfdebe7d4f3e2a1dR355-R360

Reply:
@willbarton and I tried to apply translations in the data structure, but in his words:
"at the point where QUESTIONS is defined, english (the default language) is activated"
"the url-dependent language is only activated after get_context_data is called, but becuase those questions are defined on the view, they are evaluated when the server starts up"
So we needed to find a different way of exposing the translations

They were added in `django.po` with the following comment: "# Manually added translations from within wellbeing/jinja2/wellbeing/views.py"

Reply:
I don't think they're needed, because they get translated when actually accessed for output in the template.

Reply:
@Scotchester they are wrapped in `_()` in both places for complicated reasons.

By adding `_()` around them in the view code, they get pulled into the `messages.po` file by Django's `makemessages`. We could add it by hand there, but this way Django's `makemessages` will give us line numbers, and update them when `makemessages` is run.

*BUT* at the time when the view is evaluated, the language that's "active" is English, so they are always English. Spanish is only activated when the context is fetched for the page.

By wrapping the values in the template with `_()` we ensure that the string gets translated to Spanish after Spanish is activated.

TLDR: The `_()` right here does nothing to translate the string. It's only here to make sure the string gets auto-populated in `messages.po`.

Reply:
@willbarton Two questions then:

1. Reiterating the first question of this thread: "Do the rest of the questions (and the answers) in this data structure need to be translated?"
2. If so, why am I seeing them translated, anyway? I suspect because all of these strings all already existing in `messages.po` from being `_()` wrapped when they appear in the `home.html` template?

Reply:
Oh!

I wonder — @marteki and I went back and forth on this a couple times. @marteki did we have all of them wrapped at one time? Or did you add them by hand?

`makemessages` doesn't *evaluate* the view and template, so `_(my_variable)` won't get added to `messages.po` by it.

Reply:
I do think we should wrap all of these strings so that the location they occur is recorded, but that's for human/developer/designer-friendliness, not for technical reasons.

Reply:
@willbarton - I wrapped them all originally in the view. They weren't showing up in `messages.po` at all. [The Django docs say this](https://docs.djangoproject.com/en/3.0/topics/i18n/translation/):

>(The caveat with using variables or computed values, as in the previous two examples, is that Django’s translation-string-detecting utility, django-admin makemessages, won’t be able to find these strings. More on makemessages later.)

We talked as a result. I added them by hand to `messages.po`, manually entering them with line numbers. At one point, I removed all the wraps from the view, but maybe some crept their way back in.

Reply:
Hmm, okay. I guess let's remove the last couple from the view?

Reply:
Addressed in newest commits.